discovery-provider/src/tasks/index_solana_plays.py

```diff
@@ -0,0 +1,305 @@
+        return
+
+    db = index_solana_plays.db
+
+    # Highest currently processed slot in the DB
+    latest_processed_slot = get_latest_slot(db)
+    logger.info(
+        f"index_solana_plays.py | latest used slot: {latest_processed_slot}")
+
+    # Loop exit condition
+    intersection_found = False
+
+    # List of signatures that will be populated as we traverse recent operations
+    transaction_signatures = []
+
+    # Current batch of transactions
+    transaction_signature_batch = []
+
+    last_tx_signature = None
+
+    # Traverse recent records until an intersection is found with existing Plays table
+    while not intersection_found:
+        # TODO: Is there any optimization around this limit value?
+        transactions_history = solana_client.get_confirmed_signature_for_address2(
+            TRACK_LISTEN_PROGRAM, before=last_tx_signature, limit=100
+        )
+        transactions_array = transactions_history['result']
+
+        if not transactions_array:
+            # This is considered an 'intersection' since there are no further transactions to process but
+            # really represents the end of known history for this ProgramId
+            intersection_found = True
+            logger.info(
+                f"index_solana_plays.py | No transactions found before {last_tx_signature}"
+            )
+        else:
+            with db.scoped_session() as read_session:
+                for tx in transactions_array:
+                    tx_sig = tx['signature']
+                    tx_slot = tx['slot']
+                    logger.info(
+                        f"index_solana_plays.py | Processing tx, sig={tx_sig} slot={tx_slot}"
+                    )
+                    if tx['slot'] > latest_processed_slot:
+                        transaction_signature_batch.append(tx_sig)
+                    elif tx['slot'] == latest_processed_slot:
+                        # Check the tx signature for any txs in the latest batch,
+                        # and if not present in DB, add to processing
+                        logger.info(
+                            f"index_solana_plays.py | Latest slot re-traversal\
+    slot={tx_slot}, sig={tx_sig},\
+    latest_processed_slot(db)={latest_processed_slot}")
+                        exists = get_tx_in_db(read_session, tx_sig)
+                        if exists:
+                            # Exit loop and set terminal condition since this tx has been found in DB
+                            # Transactions are returned with most recently committed first, so we can assume
+                            # subsequent transactions in this batch have already been processed
+                            intersection_found = True
+                            break
+                        else:
+                            # Ensure this transaction is still processed
+                            transaction_signature_batch.append(tx_sig)
+                    else:
```

Comment:
can we make this condition/stopping criteria a bit more explicit rather than the fall-through case? or at least document a little more. it was hard for me to reason through this a bit and understand the flow.

Reply:
yeah this condition should actually never be hit since we process every single tx, it can actually be aggregated into the above